src/overlays/actors/ovl_Bg_Haka_Huta/z_bg_haka_huta.c

```diff
@@ -31,25 +37,195 @@ const ActorInit Bg_Haka_Huta_InitVars = {
     (ActorFunc)BgHakaHuta_Update,
     (ActorFunc)BgHakaHuta_Draw,
 };
-*/
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Haka_Huta/BgHakaHuta_Init.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Haka_Huta/BgHakaHuta_Destroy.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Haka_Huta/func_8087D0AC.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Haka_Huta/func_8087D268.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Haka_Huta/func_8087D2F0.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Haka_Huta/func_8087D5B8.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Haka_Huta/func_8087D66C.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Haka_Huta/func_8087D720.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Haka_Huta/func_8087D8C0.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Haka_Huta/BgHakaHuta_Update.s")
+static InitChainEntry sInitChain[] = {
+    ICHAIN_VEC3F_DIV1000(scale, 100, ICHAIN_STOP),
+};
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Haka_Huta/BgHakaHuta_Draw.s")
+void BgHakaHuta_Init(Actor* thisx, GlobalContext* globalCtx) {
+    BgHakaHuta* this = THIS;
+    DynaPolyActor* new_var;
+    CollisionHeader* colHeader;
+
+    // Likely fake, but only way to make a match
+    new_var = &this->dyna;
+    colHeader = 0;
+    Actor_ProcessInitChain(&this->dyna.actor, sInitChain);
+    DynaPolyActor_Init(&this->dyna, 1);
+    CollisionHeader_GetVirtual(&D_06000870, &colHeader);
+    this->dyna.bgId = DynaPoly_SetBgActor(globalCtx, &globalCtx->colCtx.dyna, &this->dyna.actor, colHeader);
+    this->unk_16A = ((*new_var).actor.params >> 8) & 0xFF;
+    this->dyna.actor.params &= 0xFF;
+    if (Flags_GetSwitch(globalCtx, this->dyna.actor.params) != 0) {
```

Comment:
```suggestion
    if (Flags_GetSwitch(globalCtx, this->dyna.actor.params)) {
```
We treat Flags_GetSwitch as a boolean.